storm-core/src/jvm/org/apache/storm/utils/DisruptorQueue.java

```diff
@@ -65,7 +66,7 @@
     private static final Object INTERRUPT = new Object();
     private static final String PREFIX = "disruptor-";
     private static final FlusherPool FLUSHER = new FlusherPool();
-    private static final Timer METRICS_TIMER = new Timer("disruptor-metrics-timer", true);
+    private static final ScheduledThreadPoolExecutor METRICS_REPORTER_EXECUTOR = new ScheduledThreadPoolExecutor(1);
```

Comment:
Thanks for replacing the Timer. If the threads for this still need to be daemon threads, you should use the constructor that takes a ThreadFactory, I believe the default factory produces non-daemon threads.

Reply:
Good catch. I'll check and update as necessary.

Reply:
You were right. Fixed.